app/GioForegroundService.java

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package org.gioui;
+import android.app.Notification;
+import android.app.Service;
+import android.app.Notification;
+import android.app.Notification.Builder;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+
+public class GioForegroundService extends Service {
+	public static final String CHANNEL_ID = "GioForegroundService";
+	public static final String serviceName = "GioForeground";
+	public static final int notificationID = 42;
```

Comment:
This is likely to clash with other notification IDs. I suggest a larger (1000+) number, and maybe adding a notification ID base constant in Gio.java to make retain some control of future ID allocations.